core/src/main/java/net/opentsdb/query/filter/TagValueWildcardFilter.java

```diff
@@ -135,6 +142,84 @@ public boolean matchesAll() {
   public String[] components() {
     return components;
   }
+
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o)
+      return true;
+    if (o == null || getClass() != o.getClass())
+      return false;
+
+    // is this necessary?
+    if (!super.equals(o)) {
+      return false;
+    }
+
+    final TagValueWildcardFilter otherTagValueFilter = (TagValueWildcardFilter) o;
+
+    final boolean result = Objects.equal(has_postfix, otherTagValueFilter.has_postfix)
+            && Objects.equal(has_prefix, otherTagValueFilter.has_prefix)
+            && Objects.equal(matches_all, otherTagValueFilter.matchesAll());
+
+    if (!result) {
+      return false;
+    }
+
+
+    // comparing components
+
+    if (components != null && otherTagValueFilter.components() == null || components == null && otherTagValueFilter.components() != null) {
+      return false;
+    }
+
+    if (components != null) {
+      List<String> listComponents = Arrays.asList(components);
```

Comment:
For this one you can keep it simpler and just use the filter string. It breaks down into components but we don't need to compare those. We don't need pre or post fixes either since those are part of parsing.